modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/onboarding/upgrade/OnboardingUpgradeFeaturesPage.kt

```diff
@@ -111,69 +138,400 @@ internal fun OnboardingUpgradeFeaturesPage(
+    LaunchedEffect(pagerState) {
+        snapshotFlow { pagerState.currentPage }.collect { index ->
+            onFeatureCardChanged(index)
+        }
+    }
 
-                FeatureRow(scrollAutomatically = state.scrollAutomatically)
+    HorizontalPager(
+        pageCount = state.featureCards.size,
+        state = pagerState,
+        pageSize = PageSize.Fixed(LocalConfiguration.current.screenWidthDp.dp - 64.dp),
+        contentPadding = PaddingValues(horizontal = 32.dp),
+    ) { index ->
+        FeatureCard(
+            subscriptionFrequency = state.currentSubscriptionFrequency,
+            card = state.featureCards[index],
+            onUpgradePressed = onUpgradePressed,
+            canUpgrade = canUpgrade,
+            upgradePrice = upgradePrice,
+        )
+    }
 
-                Spacer(Modifier.weight(1f))
-                Spacer(Modifier.height(36.dp))
+    // Page indicator
+    Row(
+        Modifier
+            .height(40.dp)
+            .fillMaxWidth()
+            .padding(top = 24.dp),
+        horizontalArrangement = Arrangement.Center,
+    ) {
+        repeat(state.featureCards.size) { iteration ->
+            val color =
+                if (pagerState.currentPage == iteration) Color.White else Color.White.copy(alpha = 0.5f)
+            Box(
+                modifier = Modifier
+                    .padding(4.dp)
+                    .clip(CircleShape)
+                    .background(color)
+                    .size(8.dp)
+            )
+        }
+    }
+}
 
-                if (canUpgrade) {
-                    PlusRowButton(
-                        text = stringResource(LR.string.onboarding_upgrade_unlock_all_features),
-                        onClick = onUpgradePressed,
-                        modifier = Modifier.padding(horizontal = 24.dp),
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+fun FeatureCard(
+    subscriptionFrequency: SubscriptionFrequency,
+    card: UpgradeFeatureCard,
+    onUpgradePressed: () -> Unit,
+    canUpgrade: Boolean,
+    upgradePrice: (String) -> String,
+    modifier: Modifier = Modifier,
+) {
+    Card(
+        shape = RoundedCornerShape(24.dp),
+        elevation = 8.dp,
+        backgroundColor = Color.White,
+        modifier = modifier
+            .padding(8.dp)
+            .fillMaxWidth()
+    ) {
+        Column(
+            modifier = modifier.padding(24.dp)
+        ) {
+            Box(
+                modifier = modifier
+                    .fillMaxWidth()
+                    .padding(bottom = 12.dp),
+                contentAlignment = Alignment.TopStart
+            ) {
+                FeaturePill(card.iconRes, card.shortNameRes)
+            }
+
+            Row(
+                modifier = modifier
+                    .padding(bottom = 8.dp),
+                verticalAlignment = Alignment.Bottom,
+            ) {
+                AnimatedContent(
+                    targetState = upgradePrice(card.productIdPrefix),
```

Comment:
See internal discussion: pdeCcb-2vS-p2 regarding showing pricing on the card. You can ignore reviewing how it is updated right now.